amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/exceptions/CustomerApplicationException.java

```diff
@@ -0,0 +1,10 @@
+package software.amazon.kinesis.leases.exceptions;
+
+public class CustomerApplicationException extends Exception {
```

Comment:
nit: Would prefer to name this ApplicationException to keep this in line with the rest of the logging

Reply:
ApplicationException is a java implemented exception. Trying to differentiate from it.